conan/tools/premake/premake.py

```diff
@@ -0,0 +1,39 @@
+class Premake(object):
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+
+    # automatically chooses premake action based on used compiler
+    def configure(self):
+        if "Visual Studio" in self.settings.compiler:
```

Comment:
This is typically done in ``PremakeToolchain`` that is called at ``generate()`` time.
This is the feature that allows using ``Premake`` as developer too ``conan install`` + ``premake``...

Good enough for this first iteration, but something to do before making this integration stable.

Reply:
from what i understood `[tool]Toolchain` is for setting up configuration and `[tool]` is for helper functions to call the tool from python.
unlike CMake Premake doesn't have separate "configure" and "generate" steps, and it takes the target build system as a command line argument when called.
so it kinda falls into both categories of config in the sense of choosing build system and helper for calling premake.
I think it makes more sense for it to belong with the helper because that's the way premake is intended to be used (it doesn't look for config files when called).

Reply:
Understood, yes, I assumed by a very quick read that it was using something like vcvars, but it is not the case. Looking good.